Marlin/pins_ZRIB_2.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#if HOTENDS > 2
+  #error "Zrib v2 supports up to 2 hotends. Comment this line to keep going."
+#endif
+
+#define BOARD_NAME "ZRIB v2"
+
+#include "pins_RAMPS.h"
+
+#undef FAN1_PIN
+#define FAN1_PIN 6
+
+#if HOTENDS == 2
+  #undef HEATER_1_PIN
+  #define HEATER_1_PIN 7
+#endif
```

Comment:
With 2 hotends, the RAMPS variant is either EEF or EEB. In both cases, `HEATER_1_PIN` is assigned the value of `RAMPS_D9_PIN`. So, it will be more flexible to put this line above the `#include` instead:
```cpp
#define RAMPS_D9_PIN 7
```
…or possibly…
```
#if HOTENDS == 2
  #define RAMPS_D9_PIN 7
#endif
…because with `HOTENDS == 1` the D9 pin is given to `FAN_PIN`.
```
The code in `pins_RAMPS.h` that decides how to allocate the MOSFETs begins:
```cpp
#if HOTENDS > 1
  #if TEMP_SENSOR_BED
    #define IS_RAMPS_EEB
  #else
    #define IS_RAMPS_EEF
  #endif
#elif TEMP_SENSOR_BED
  #define IS_RAMPS_EFB
#else
  #define IS_RAMPS_EFF
#endif
```
…and then below that are the actual MOSFET allocations…